src/sentry/sentry_metrics/querying/api.py

```diff
@@ -0,0 +1,304 @@
+            totals[metric_name] = metric_values[1]
+
+        inner_group = {
+            "by": {name: value for name, value in group_key},
+            "series": series,
+            "totals": totals,
+        }
+
+        translated_groups.append(inner_group)
+
+    translated_meta = [
+        {"name": meta_name, "type": meta_type} for meta_name, meta_type in intermediate_meta.items()
+    ]
+
+    return {
+        "intervals": intervals,
+        "groups": translated_groups,
+        "meta": translated_meta,
+        "start": start,
+        "end": end,
+    }
+
+
+def _build_request(organization: Organization, query: MetricsQuery) -> Request:
+    dataset = (
+        Dataset.Metrics
+        if query.scope.use_case_id == UseCaseID.SESSIONS.value
+        else Dataset.PerformanceMetrics
+    )
+    return Request(
+        dataset=dataset.value,
+        query=query,
+        app_id="default",
+        tenant_ids={"referrer": "metrics.data", "organization_id": organization.id},
+    )
+
+
+def _infer_use_case_id_from_timeseries(timeseries: Timeseries) -> UseCaseID:
+    if timeseries.metric.mri is not None:
+        mri = timeseries.metric.mri
+    else:
+        mri = get_mri(timeseries.metric.public_name)
+
+    try:
+        return extract_use_case_id(mri)
+    except ValidationError:
+        raise InvalidParams(f"The query contains an invalid MRI: {mri}")
+
+
+def _execute_series_and_totals_query(
+    organization: Organization, interval: int, base_query: MetricsQuery
+) -> Mapping[str, Any]:
+    # We infer the use case id from the timeseries.
+    inferred_use_case_id = _infer_use_case_id_from_timeseries(base_query.query)
+    extended_scope = base_query.scope.set_use_case_id(inferred_use_case_id.value)
+    base_query = base_query.set_scope(extended_scope)
+
+    # First we make the series query.
+    query = base_query.set_rollup(Rollup(interval=interval))
+    series_result = run_query(request=_build_request(organization, query))
+
+    # Second we make the totals query by taking the same modified interval computed by the series query.
+    modified_start = series_result["modified_start"]
+    modified_end = series_result["modified_end"]
+    query = (
+        base_query.set_start(modified_start).set_end(modified_end).set_rollup(Rollup(totals=True))
+    )
+    totals_result = run_query(request=_build_request(organization, query))
+
+    return {**series_result, "totals": totals_result["data"]}
+
+
+def run_metrics_query(
+    fields: Sequence[str],
+    query: Optional[str],
+    group_bys: Optional[Sequence[str]],
+    interval: int,
+    start: datetime,
+    end: datetime,
+    organization: Organization,
+    projects: Sequence[Project],
+):
+    base_scope = MetricsScope(
+        org_ids=[organization.id],
+        project_ids=[project.id for project in projects],
+    )
+
+    # Build the basic query that contains the metadata.
+    base_query = MetricsQuery(
+        filters=_parse_filters(query) if query else None,
```

Comment:
Enoch has removed the filters and groupby at the `MetricsQuery` layer in a future version of the SDK. So we will only support them on the `Timeseries` directly. If its possible to incorporate that now, it will make our lives easier in the future.

Reply:
I will change that